API/Repositories/Implementation/TestsRepository.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using API.Data;
+using API.Dtos.Test;
+using API.Entities;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Repositories.Implementation
+{
+    public class TestsRepository : ITestsRepository
+    {
+        private CourseContext _context;
+        private IMapper _mapper;
+        public TestsRepository(
+            CourseContext context,
+            IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<Result<List<GetTestDto>>> GetTestsOfLesson(int lessonId)
+        {
+            var dbTests = _context.Tests
+                .Include(t => t.Options)
+                .Where(t => t.Lesson.Id == lessonId);
+
+            if (dbTests == null || !dbTests.Any())
+            {
+                return new Result<List<GetTestDto>> { IsSuccess = false, ErrorMessage = "Tests for this lesson not found." };
```

Comment:
Consider the option of creating a _ErrorMessages_ list that will store all error texts.

Reply:
I think one error message will be enough for me. Although in other cases it would really be appropriate to output a list of error messages. Thanks for your suggestion